util/sqlexec/restricted_sql_executor.go

```diff
@@ -89,6 +90,10 @@ type SQLExecutor interface {
 	// ExecuteInternal means execute sql as the internal sql.
 	ExecuteInternal(ctx context.Context, sql string, args ...interface{}) (RecordSet, error)
 	ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (RecordSet, error)
+	// allowed when tikv disk full happened.
+	SetDiskFullOpt(level kvrpcpb.DiskFullOpt)
```

Comment:
I saw only transactions need DiskFullOpt, why do you add the functions to so many interfaces?

Reply:
We need doInsert set this flag, and uesd SQLExecutor set interface.